aten/src/ATen/native/Indexing.cpp

```diff
@@ -309,7 +305,7 @@ Tensor & index_copy_(Tensor & self, int64_t dim, const Tensor & index, const Ten
   }
   auto sourceSlicedSizes = std::vector<int64_t>(source.sizes());
   if (sourceSlicedSizes.size() > 0) {
-    sourceSlicedSizes.erase(sourceSlicedSizes.begin());
+    sourceSlicedSizes.erase(sourceSlicedSizes.begin() + dim);
```

Comment:
Which test is for this fix? (For my education.)

Reply:
I'm not sure exactly what the question is -- are you asking which new test case corresponds to this?  In that case, it's any index_copy that doesn't use dim 0.